packages/denylist/src/denylist.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-env serviceworker, browser */
+
+import { normalizeCid } from './utils/cid'
+import { getFromDenyList } from './utils/denylist'
+import { JSONResponse } from '@web3-storage/worker-utils/response'
+
+/**
+ * @param {Array<string>} params
+ * @param {(params: Array<string>, request: Request, env: import('./env').Env) => Promise<Response>} fn
+ * @returns {import('itty-router').RouteHandler<Request>}
+ */
+function withRequiredQueryParams (params, fn) {
+  /**
+   * @param {Request} request
+   * @param {import('./env').Env} env
+   */
+  return async function (request, env) {
+    const searchParams = (new URL(request.url)).searchParams
+
+    for (const paramName of params) {
+      const paramValue = searchParams.get(paramName)
+      if (!paramValue) {
+        return new Response(`${paramName} is a required query param`, { status: 400 })
+      }
+
+      if (paramName === 'cid') {
+        try {
+          await normalizeCid(paramValue)
+        } catch (e) {
+          return new Response('cid query param is invalid', { status: 400 })
+        }
+      }
+    }
+
+    return await fn(params.map(param => String(searchParams.get(param))), request, env)
+  }
+}
+
+export const denylistGet = withRequiredQueryParams(['cid'],
+  /**
+   * Returns badbits denylist results.
+   */
+  async function (params, request, env) {
+    const [cid] = params
+
+    const denyListResource = await getFromDenyList(cid, env)
+    if (denyListResource) {
+      try {
+        return new JSONResponse(JSON.parse(denyListResource), { status: 200 })
```

Comment:
I agree with making this 200 + 404 👍🏼  and we pass what we know from the CID, letting consumers take needed actions